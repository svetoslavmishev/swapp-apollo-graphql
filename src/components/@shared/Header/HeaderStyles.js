import { makeStyles } from '@material-ui/styles';

const styles = makeStyles(theme => ({
  appBar: ({ currentTheme }) => theme.palette[currentTheme].AppBar,
  toolBar: {
    display: 'flex',
    justifyContent: 'flex-end',
    alignItems: 'center',
    cursor: 'pointer'
  },
  toolBarLeft: {
    display: 'flex',
    justifyContent: 'flex-end',
    alignItems: 'center',
    width: '100%',
    '& a': {
      padding: 8,
      textDecoration: 'none',
      color: ({ currentTheme }) => theme.palette[currentTheme].AppBar.fontColor
    },
    '&:hover svg': {
      cursor: 'pointer'
    }
  }
}));

export default styles;
