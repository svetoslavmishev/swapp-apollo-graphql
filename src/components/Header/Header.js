import React, { useContext } from 'react';
import { AppBar, Toolbar, Button, Typography } from '@material-ui/core';
import styles from './HeaderStyles';
import { ThemeContext } from '../../themeContext';

const Header = () => {
  const { currentTheme, toggleTheme } = useContext(ThemeContext);
  const classes = styles({ currentTheme });

  return (
    <>
      <AppBar position="static" className={classes.appBar}>
        <Toolbar>
          <div className={classes.toolBar} onClick={toggleTheme}>
            SWAPP
          </div>
          <div
            style={{
              display: 'flex',
              justifyContent: 'flex-end',
              alignItems: 'center',
              width: '100%'
            }}
            className={classes.toolBarLeft}
          >
            <Typography>Episodes</Typography>
            <Typography>Characters</Typography>
            <Button color="inherit">Logout</Button>
          </div>
        </Toolbar>
      </AppBar>
    </>
  );
};

export default Header;
