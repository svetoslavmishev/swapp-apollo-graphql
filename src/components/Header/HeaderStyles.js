import { makeStyles } from '@material-ui/styles';

const styles = makeStyles(theme => ({
  appBar: ({ currentTheme }) => theme.palette[currentTheme].AppBar,
  toolBar: {
    display: 'flex',
    justifyContent: 'flex-end',
    alignItems: 'center',
    cursor: 'pointer',
    '&:div > p': {
      background: 'red'
    }
  }
}));

export default styles;
