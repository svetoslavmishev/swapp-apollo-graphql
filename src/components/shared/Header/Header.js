import React, { useContext } from 'react';
import { useApolloClient } from '@apollo/react-hooks';
import { Link } from 'react-router-dom';
import { AppBar, Toolbar } from '@material-ui/core';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import styles from './HeaderStyles';
import { ThemeContext } from '../../../themeContext';

const Header = () => {
  const { currentTheme, toggleTheme } = useContext(ThemeContext);
  const client = useApolloClient();
  const classes = styles({ currentTheme });
  const handleLogout = () => {
    client.writeData({ data: { isAuthenticated: false } });
    localStorage.clear();
  };

  return (
    <AppBar position="static" className={classes.appBar}>
      <Toolbar>
        <div
          className={`${classes.toolBar} ${classes.logo}`}
          onClick={toggleTheme}
        >
          SWAPP
        </div>
        <div className={classes.toolBarLeft}>
          <Link to={'/episodes'}>Episodes</Link>
          <Link to={'/characters'}>Characters</Link>
          <div className="sm:pr-2">
            <ExitToAppIcon onClick={handleLogout} />
          </div>
        </div>
      </Toolbar>
    </AppBar>
  );
};

export default Header;
