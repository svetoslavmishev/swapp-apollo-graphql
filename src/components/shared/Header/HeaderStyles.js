import { makeStyles } from '@material-ui/styles';

const styles = makeStyles(theme => ({
  appBar: ({ currentTheme }) => theme.palette[currentTheme].AppBar,
  toolBar: {
    fontFamily: 'StarWars',
    fontSize: 24,
    display: 'flex',
    justifyContent: 'flex-end',
    alignItems: 'center',
    cursor: 'pointer'
  },
  toolBarLeft: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'flex-end',
    alignItems: 'center',
    width: '100%',
    '& a': {
      padding: 8,
      color: ({ currentTheme }) => theme.palette[currentTheme].AppBar.fontColor
    },
    '&:hover a, svg': {
      cursor: 'pointer'
    }
  },
  logo: {
    color: theme.palette.commonColors.yellow
  }
}));

export default styles;
