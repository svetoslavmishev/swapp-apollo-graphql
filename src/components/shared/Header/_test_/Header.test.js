import React from 'react';
import { shallow } from 'enzyme';
import { ThemeContext, theme } from '../../../../themeContext';
import Header from '../Header';

describe('<Header />', () => {
  it('renders <Header /> without error', () => {
    const wrapper = shallow(
      <ThemeContext.Provider value={theme}>
        <Header />
      </ThemeContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
