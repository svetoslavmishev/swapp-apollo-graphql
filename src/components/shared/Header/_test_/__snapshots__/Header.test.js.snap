// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders <Header /> without error 1`] = `ShallowWrapper {}`;
